shared/types/genesis.go

```diff
@@ -0,0 +1,74 @@
+package types
```

Comment:
This makes sense to move it here IMO - but we need a better clarification of testing vs real structures. TODO

Reply:
Maybe each package should declare a "Genesis" interface and we have a single shared object containing all namespaced implementations of these genesis interfaces at the `shared` level?

Reply:
Sorry, genesis related files were out of scope for this PR and some lack of git-foo made the original PR bigger than necessary.